src/shogun-core-main/src/main/java/de/terrestris/shoguncore/util/interceptor/secure/WfsResponseInterceptor.java

```diff
@@ -180,6 +185,31 @@ public Response interceptDescribeFeatureType(MutableHttpServletRequest request,
         LOG.debug("Intercepting WFS DescribeFeatureType response");
         ByteArrayOutputStream bout = new ByteArrayOutputStream();
         try {
+            if (response.getHeaders().get("content-type").get(0).toLowerCase(Locale.ROOT).contains("json")) {
+                ObjectMapper objectMapper = new ObjectMapper();
+                JsonNode jsonObject = objectMapper.readTree(response.getBody());
+                String ns = jsonObject.get("targetPrefix").asText();
+                ArrayNode featureTypes = (ArrayNode) jsonObject.get("featureTypes");
+                List<Layer> allowedLayers = layerService.findAll();
+                for (int i = 0; i < featureTypes.size(); i++) {
+                    String typeName = ns + ":" + featureTypes.get(i).get("typeName").asText();
+                    boolean match = false;
+                    for (Layer allowedLayer : allowedLayers) {
+                        if (allowedLayer.getSource() instanceof TileWmsLayerDataSource) {
+                            String names = ((TileWmsLayerDataSource) allowedLayer.getSource()).getLayerNames();
+                            if (names.equalsIgnoreCase(typeName)) {
+                                match = true;
+                            }
```

Comment:
The loop can be exited here to gain some performance (`.getSource()` will be fetched lazy ASFAIK).

Reply:
done